File: Data Structures & Algorithms/is-palindrome/submission-0.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isPalindrome(s) {
        let filteredStr = '';
        
        // Build the filtered string with alphanumeric characters only
        for (let i = 0; i < s.length; i++) {
            let char = s[i].toLowerCase();
            if ((char >= 'a' && char <= 'z') || (char >= '0' && char <= '9')) {
                filteredStr += char;
            }
        }
        
        // Check if the filtered string is a palindrome
        let left = 0;
        let right = filteredStr.length - 1;
        
        while (left < right) {
            if (filteredStr[left] !== filteredStr[right]) {
                return false;
            }
            left++;
            right--;
        }
        
        return true;
    }
}

File: Data Structures & Algorithms/valid-palindrome-ii/submission-0.go
func validPalindrome(s string) bool {
	left,right:=0,len(s)-1
	for left< right {
		if s[left]!=s[right] {
			return isPalindrome(s, left, right-1) || isPalindrome(s, left+1, right)
		}
		left++
		right--
	}

	return true 
}

func isPalindrome(s string, l int, r int) bool {

	for l<r {
		if s[l]!=s[r] {
			return false
		}
		l++
		r--
	}

	return true
}